Refactor(eos_validate_state): Keep only ANTA mode in eos_validate_state (#4286)
aristanetworks/ansible-avd: ansible_collections/arista/avd/docs/release-notes/5.x.x.md

TODO: add level4 sections for breaking changes outlining the change and referring to porting guide for migration.

### Breaking or behavioral changes in eos_validate_state

Starting AVD 5.0.0, `eos_validate_state` uses ANTA as backend, which will change the default behavior (same as setting `use_anta: true` in the latest AVD 4.x.x versions).
If you have not been using `use_anta: true` in latest versions of AVD 4.x.x, please go through the following sections for details on the changes.

Please report any issues on the GitHub [discussions board](https://github.com/aristanetworks/avd/discussions).

Breaking changes may require modifications to the inventory, input variables or playbooks. See the [Porting guide for AVD 5.x.x](../porting-guides/5.x.x.md)
for details.

#### Loose mode to ignore playbook error is no longer supported

> No porting guide entry.

Reviewer: Instead maybe add a small description on how errors are handled in the new integration? Like an unreachable device.

#### Input for hardware tests have been updated

> The inputs for the hardware test have changed. [Porting guide for AVD 5.x.x](../porting-guides/5.x.x.md#input-for-hardware-tests-have-been-updated)

Reviewer: ```suggestion
The inputs for the hardware test have changed. See the [porting guide](../porting-guides/5.x.x.md#input-for-hardware-tests-have-been-updated) for details.
```

#### Filtering tests using Ansible tags support was removed

> Instead use the `skip_tests` mechanism. [Porting guide for AVD 5.x.x](../porting-guides/5.x.x.md#filtering-tests-using-ansible-tags-support-was-removed)

Reviewer: ```suggestion
Instead use the `skip_tests` mechanism. See the [porting guide](../porting-guides/5.x.x.md#filtering-tests-using-ansible-tags-support-was-removed) for details.
```

#### Changes to CSV and Markdown reports

- Hardware tests are now collapsed.
- Sorting of test results is now done per device as opposed to per category.
- Test categories, descriptions and inputs have been improved to follow ANTA.
- CSV report headers updated from `test_id,node,test_category,test_description,test,result,failure_reason` to `id,dut,categories,test,description,inputs,result,messages`
- Tests skipped by ANTA will be marked as `SKIPPED` in the final reports.

> No porting guide entry.
> 

Reviewer: ```suggestion
```

#### Undeployed devices

- All tests will be removed from the catalog for a device flagged as undeployed using the host level variable [`is_deployed: false`](https://avd.sh/en/stable/roles/eos_designs/docs/input-variables.html#flagging-a-device-as-not-deployed). Additionally, all tests take into account the `is_deployed` variable value and remove tests accordingly.

!!! warning
    Tests can also be automatically removed from the catalogs depending on the structured configuration of the devices. ANTA is therefore not aware of these tests and they will not appear in the final report. For example, the `AvdTestMLAG` tests will not be present in the test catalog of a device that does not have an MLAG configuration in its structured configuration.

#### BGP test change

- BGP tests will only run if `service_routing_protocols_model` is set to `multi-agent` in the structured configuration file.

> !!! note

Reviewer: Some later version even removed the option for ribd

Author: Ack - not sure what to do with this comment?

>     Starting from version 4.30.1F, `service_routing_protocols_model` is preset to `multi-agent` by default on EOS devices.

Reviewer: ```suggestion
    Starting from version 4.30.1F, `service_routing_protocols_model` is set to `multi-agent` by default on EOS devices.
```

TODO: add level4 sections for breaking changes outlining the change and referring to porting guide for migration.

### Breaking or behavioral changes in eos_validate_state

Starting AVD 5.0.0, `eos_validate_state` uses ANTA as backend, which will change the default behavior (same as setting `use_anta: true` in older versions).

> If you have not been using `use_anta: true` in latest versions of AVD 4.x.x, you should expect faster execution, and if not please report on the GitHub [discussions board](https://github.com/aristanetworks/avd/discussions)

Reviewer: ```suggestion
If you have not been using `use_anta: true` in the latest versions of AVD 4.x.x, please go through the following sections for details on the changes.

Please report any issues on the GitHub [discussions board](https://github.com/aristanetworks/avd/discussions).
```

Author: this removes the point of execution being faster though

for details.

TODO: add level4 sections for breaking changes outlining the change and referring to porting guide for migration.

### Breaking or behavioral changes in eos_validate_state

Starting AVD 5.0.0, `eos_validate_state` uses ANTA as backend, which will change the default behavior (same as setting `use_anta: true` in the latest AVD 4.x.x versions).
If you have not been using `use_anta: true` in latest versions of AVD 4.x.x, please go through the following sections for details on the changes.

Please report any issues on the GitHub [discussions board](https://github.com/aristanetworks/avd/discussions).

Breaking changes may require modifications to the inventory, input variables or playbooks. See the [Porting guide for AVD 5.x.x](../porting-guides/5.x.x.md)
for details.

> #### Loose mode to ignore playbook error is no longer supported

Author: ```suggestion
#### Loose mode is the only supported mode
```

> No porting guide entry.

Author: ```suggestion
In the previous implementation, loose mode was used by default to avoid stopping the playbook and hence the tests when they were run task by task in Ansible. It was possible to turn loose mode off with a variable that would make the playbook stop at the first failing tasks for whatever reason on each host.

Loose mode is now the default and all the tests are always run and the errors are all present in the final report.

- No porting guide entry.
```
